src/main/kotlin/network/warzone/mars/player/achievements/AchievementEmitter.kt

```diff
@@ -35,13 +35,18 @@ class AchievementEmitter(private val achievement: Achievement) {
             println("Achievement " + achievement.name + " earned by " + player.name)
         }
 
-        // Emit achievement completion to the database.
-        ApiClient.emit(
-            OutboundEvent.PlayerAchievement, PlayerAchievementData(
-                SimplePlayer(profile._id, profile.name),
-                achievement._id,
-                Date().time
-            )
+        val data = PlayerAchievementData(
+            SimplePlayer(profile._id, profile.name),
+            achievement._id,
+            Date().time
         )
+
+        if (achievement.firstCompletion.toString() == "00000000-0000-0000-0000-000000000000") {
```

Comment:
Mentioned in API review but should do a null check instead of checking for an empty UUID